README.md

```diff
@@ -320,6 +320,56 @@ describe('MockRender', () => {
 });
 ```
 
+## MockProvider
+
+Provides simple way to providers/services when including them in the testing module. 
+
+By default provider will be mocked with an empty method, but if you provide factory method for mocking, all methods will be mocked with that setting. This is useful for example when we want to mock service methods with something like `jasmine.createSpy` or `jest.fn()`.
+
+### Usage Example
+```typescript
+describe('MockProvider', () => {
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        MockProvider(RealService)
+      ]
+    });
+
+    const service = TestBed.get(RealService);
+  });
+});
+```
+
+With factory method:
+```typescript
+describe('MockProvider', () => {
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        MockProvider(RealService, (methodName) => jasmine.createSpy(methodName))
```

Comment:
Maybe this could happen automatically? Or you need to pass factory method for spy because when one uses JEST the way the spy is created differs? Makes sense :-)

Reply:
Indeed, to support different testing frameworks this makes sense.